src/renderer/components/ImageCropper/index.tsx

```diff
@@ -55,79 +70,116 @@ export default function ImageCropper({
   const posX = useRef<number>(0)
   const posY = useRef<number>(0)
 
+  // multiplier to flip direction on X or Y
+  // either positive or negative
+  const flipDirX = useRef<number>(1)
+  const flipDirY = useRef<number>(1)
+
   const targetWidth = useRef<number>(desiredWidth)
   const targetHeight = useRef<number>(desiredHeight)
 
   const containerScale = useRef<number>(1.0)
   const zoom = useRef<number>(1.0)
   const initialZoom = useRef<number>(1.0)
+  /** we use degrees for simple comparision check */
+  const rotation = useRef<number>(0)
 
-  const onSubmit = () => {
+  const onSubmit = async () => {
     if (!fullImage.current) {
       return
     }
 
-    const canvas = document.createElement('canvas')
+    const canvas = tmpCanvas.current
+
+    if (!canvas) {
+      return
+    }
+
+    const resultW = targetWidth.current / zoom.current
+    const resultH = targetHeight.current / zoom.current
     canvas.width = targetWidth.current
     canvas.height = targetHeight.current
 
     const context = canvas.getContext('2d', {
       willReadFrequently: false,
     }) as CanvasRenderingContext2D
-    context.imageSmoothingEnabled = false
 
-    const imgW = fullImage.current.clientWidth
-    const imgH = fullImage.current.clientHeight
+    context.imageSmoothingEnabled = false
 
+    context.translate(canvas.width / 2, canvas.height / 2)
+    context.rotate((rotation.current * Math.PI) / 180)
+    // to flip we need not only negative width or height in drawImage
+    // but also a context scale
+    context.scale(flipDirX.current, flipDirY.current)
     context.drawImage(
       fullImage.current as CanvasImageSource,
-      posX.current - (targetWidth.current / zoom.current - imgW) / 2,
-      posY.current - (targetHeight.current / zoom.current - imgH) / 2,
-      targetWidth.current / zoom.current,
-      targetHeight.current / zoom.current,
-      0,
-      0,
-      targetWidth.current,
-      targetHeight.current
+      posX.current - Math.floor(resultW / 2) * flipDirX.current,
+      posY.current - Math.floor(resultW / 2) * flipDirY.current,
+      // negative values flip the image
+      resultW * flipDirX.current,
+      resultH * flipDirY.current,
+      canvas.width / -2,
+      canvas.height / -2,
+      canvas.width,
+      canvas.height
     )
-    ;(async () => {
-      const tempfilename = `profile_pic_${Date.now()}.png`
-      const tempfilepath = await runtime.writeTempFileFromBase64(
-        tempfilename,
-        canvas.toDataURL('image/png').split(';base64,')[1]
-      )
 
-      setLastPath(dirname(filepath))
-      onResult(tempfilepath)
-      onClose()
-    })()
+    const tempfilename = `profile_pic_${Date.now()}.png`
+    const tempfilepath = await runtime.writeTempFileFromBase64(
+      tempfilename,
+      canvas.toDataURL('image/png').split(';base64,')[1]
```

Comment:
I don't think it's a good idea to convert any image to PNG. Normal photography images should be something like JPG as converting them to PNG will drastically increase their size.

Reply:
afaik, core scales down images to a few kb. but that needs to be double checked. not totally sure therefore if PNG or JPG is better here, if sizes are some 1000x1000 pixels, JPG is probably still better.

@iequidoo did lots of things in that area and will know :)

Reply:
> afaik, core scales down images to a few kb. but that needs to be double checked. not totally sure therefore if PNG or JPG is better here, if sizes are some 1000x1000 pixels, JPG is probably still better.
> 
> @iequidoo did lots of things in that area and will know :)

We've discussed that with Simon, decided to stay with PNG since it's lossless. Right now the result is limited to 256x256 so savings on size I believe are pretty small.